src/som/compiler/Parser.java

```diff
@@ -1444,6 +1450,29 @@ private LiteralNode literalString() throws ParseError {
     return new StringLiteralNode(s, getSource(coord));
   }
 
+  private LiteralNode literalArray(final MethodBuilder builder) throws ProgramDefinitionError {
+    SourceCoordinate coord = getCoordinate();
+    List<ExpressionNode> expressions = new ArrayList<ExpressionNode>();
+
+    expect(NewLiteralArray, DelimiterOpeningTag.class);
+
+    while (true) {
+      comments();
+
+      switch (sym) {
+        case EndLiteralArray:
+          expect(EndLiteralArray, DelimiterClosingTag.class);
+          return createLiteralArray(expressions, getSource(coord));
+        case Period:
```

Comment:
Agreed, that suggestion is much more clear. I've made the change, although when you say _roughly_ perhaps you meant _exactly_.